packages/amplication-git-service/src/providers/githubProvider.ts

```diff
@@ -0,0 +1,323 @@
+import { Octokit } from 'octokit';
+import { GitProvider } from './git-provider.interface';
+import { PullRequestMeta } from '../Dto/entities/pull-request-meta.dto';
+
+export class GithubProvider implements GitProvider {
+  constructor(
+    private octokit: Octokit,
+    private owner: string,
+    private repo: string
+  ) {}
+
+  private static async createFilesTree(
+    octokit: Octokit,
+    owner: string,
+    repo: string,
+    latestCommitSha: string,
+    files: { path: string; content: string }[]
+  ): Promise<string> {
+    return (
+      await octokit.rest.git.createTree({
+        owner,
+        repo,
+        tree: files.map(file => {
+          return {
+            ...file,
+            type: 'blob',
+            mode: '100644'
+          };
+        }),
+        base_tree: latestCommitSha
+      })
+    ).data.sha;
+  }
+
+  private static async createCommit(
+    octokit: Octokit,
+    owner: string,
+    repo: string,
+    message: string,
+    latestCommitSha: string,
+    fileTreeSha: string
+  ): Promise<string> {
+    return (
+      await octokit.rest.git.createCommit({
+        owner,
+        repo,
+        message,
+        tree: fileTreeSha,
+        parents: [latestCommitSha]
+      })
+    ).data.sha;
```

Comment:
I think it would be more readable:
```ts
const { data } = await octokit.rest.git.createCommit({
  owner,
  repo,
  message,
  tree: fileTreeSha,
  parents: [latestCommitSha]
});

return data.sha;
```

I don't like abusing `return (somePromise()).x.y.z` as it compromises on readability.

Reply:
Also, what about error handling?

Reply:
currently, I'm ok with this handled in higher levels

Reply:
The typescript compiler doesn't like it :(